File: test/python/test_mongo.py
"""MongoDB 接続テスト（PyMongo）。"""
from pymongo import MongoClient

from config import CONNECT_TIMEOUT_MS, MONGO
from util import print_table


def run():
    print("===== Test MongoDB =====")
    # PyMongo は接続を遅延させるため、実際の待ち時間はサーバ選択タイムアウトで決まる。
    # 接続タイムアウトだけでは既定の 30 秒待ってしまうので、両方を指定する。
    client = MongoClient(
        MONGO["uri"],
        connectTimeoutMS=CONNECT_TIMEOUT_MS,
        serverSelectionTimeoutMS=CONNECT_TIMEOUT_MS,
    )
    try:
        client.admin.command("ping")
        print("Connected successfully to server.")

        tbl = client[MONGO["db"]][MONGO["collection"]]

        # Delete
        deleted = tbl.delete_many({}).deleted_count
        print(f"{deleted} document(s) deleted")

        # Insert
        tbl.insert_many(
            [
                {"name": "Dan", "age": 18},
                {"name": "Bob", "age": 22},
                {"name": "John", "age": 30},
            ]
        )

        # Select
        docs = list(tbl.find({}, {"_id": 0}))
        print_table(["name", "age"], [(d["name"], d["age"]) for d in docs])
    finally:
        client.close()


if __name__ == "__main__":
    run()

File: test/dotnet/TestMongo.cs
using MongoDB.Bson;
using MongoDB.Driver;

namespace DbTests;

// MongoDB 接続テスト（MongoDB.Driver）。
public static class TestMongo
{
    public static async Task RunAsync()
    {
        Console.WriteLine("===== Test MongoDB =====");
        // ドライバは接続を遅延させるため、実際の待ち時間はサーバ選択タイムアウトで決まる。
        // 接続タイムアウトだけでは既定の 30 秒待ってしまうので、両方を指定する。
        var settings = MongoClientSettings.FromConnectionString(Config.MongoUri);
        settings.ConnectTimeout = TimeSpan.FromSeconds(Config.ConnectTimeoutSec);
        settings.ServerSelectionTimeout = TimeSpan.FromSeconds(Config.ConnectTimeoutSec);
        var client = new MongoClient(settings);
        await client.GetDatabase("admin").RunCommandAsync<BsonDocument>(new BsonDocument("ping", 1));
        Console.WriteLine("Connected successfully to server.");

        var coll = client.GetDatabase(Config.MongoDb).GetCollection<BsonDocument>(Config.MongoCollection);

        // Delete
        var deleted = (await coll.DeleteManyAsync(Builders<BsonDocument>.Filter.Empty)).DeletedCount;
        Console.WriteLine($"{deleted} document(s) deleted");

        // Insert
        await coll.InsertManyAsync(new[]
        {
            new BsonDocument { { "name", "Dan" }, { "age", 18 } },
            new BsonDocument { { "name", "Bob" }, { "age", 22 } },
            new BsonDocument { { "name", "John" }, { "age", 30 } },
        });

        // Select
        var docs = await coll.Find(Builders<BsonDocument>.Filter.Empty).ToListAsync();
        Table.Print(
            new[] { "name", "age" },
            docs.Select(d => new object?[] { d["name"].AsString, d["age"].AsInt32 }).ToList());
    }
}
